### PR TITLE
More spec compliant `Blob.prototype.type`

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
- Fixes a couple test failures
- Adds a MIMEType hash table to intern mime type strings so we usually won't have to allocate them
- Blob.prototype.type with a non-ascii value should return ""
- Uses saturated and/or wrapping arithmetic in .slice() to more clearly handle integer overflow
- Checks isNumber in a couple places we should've been checking

### CODE REVIEW

File: src/bun.js/webcore/blob.zig [hunk 1]
```diff
@@ -2410,51 +2410,84 @@ pub const Blob = struct {
         // If the optional end parameter is not used as a parameter when making this call, let relativeEnd be size.
         var relativeEnd: i64 = @intCast(i64, this.size);
 
+        if (args.ptr[0].isString()) {
```
Comment (dylan-conway) [inline src/bun.js/webcore/blob.zig:2413]: needs a 0 length check, `blob.slice()` will crash

Comment (Jarred-Sumner) [reply src/bun.js/webcore/blob.zig:2413]: `callframe.arguments(3)` presets everything to to `JSValue.zero` and .ptr avoids the bounds check

Comment (dylan-conway) [reply src/bun.js/webcore/blob.zig:2413]: `isString()` will crash if jsvalue is zero

Comment (dylan-conway) [reply src/bun.js/webcore/blob.zig:2413]: `blob.slice(number)` also has the same problem. maybe we should have a check for zero in `isString()`

Comment (Jarred-Sumner) [reply src/bun.js/webcore/blob.zig:2413]: fixed

**Pull request merged**